calculator_test.go

```diff
@@ -2,23 +2,304 @@ package calculator_test
+			n1:    -3,
+			n2:    -5,
+			extra: nil,
+			want:  2,
+		},
+		{
+			name:  "Substract five numbers",
+			n1:    20,
+			n2:    5,
+			extra: []float64{4, 3, 7},
+			want:  1,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got := calculator.Subtract(tC.n1, tC.n2, tC.extra)
+			if tC.want != got {
+				t.Errorf("want %f, got %f", tC.want, got)
+			}
+		})
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	testCases := []struct {
+		name  string
+		want  float64
+		n1    float64
+		n2    float64
+		extra []float64
+	}{
+		{
+			name:  "Multiply two positive numbers returns positive",
+			n1:    3,
+			n2:    2,
+			extra: nil,
+			want:  6,
+		},
+		{
+			name:  "Multiply positive and negative should return negative",
+			n1:    3,
+			n2:    -2,
+			extra: nil,
+			want:  -6,
+		},
+		{
+			name:  "Multiply two negative numbers returns positive",
+			n1:    -3,
+			n2:    -2,
+			extra: nil,
+			want:  6,
+		},
+		{
+			name:  "Multiply fractions return fraction",
+			n1:    3.2,
+			n2:    2.501,
+			extra: nil,
+			want:  8.0032,
+		},
+		{
+			name:  "Multiply by zero return zero",
+			n1:    1943,
+			n2:    0,
+			extra: nil,
+			want:  0,
+		},
+		{
+			name:  "Multiply five numbers",
+			n1:    2,
+			n2:    2,
+			extra: []float64{5, 3, 2},
+			want:  120,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got := calculator.Multiply(tC.n1, tC.n2, tC.extra)
+			if tC.want != got {
+				t.Errorf("want %f, got %f", tC.want, got)
+			}
+		})
 	}
 }
 
-// func TestSubtract(t *testing.T) {
-// 	t.Parallel()
-// 	var want float64 = 2
-// 	got := calculator.Subtract(4, 2)
-// 	if want != got {
-// 		t.Errorf("want %f, got %f", want, got)
-// 	}
-// }
+// It only consider errExpected variable if an error occours.
+func TestDivide(t *testing.T) {
+	testCases := []struct {
+		name        string
+		errExpected bool
+		want        float64
+		n1          float64
+		n2          float64
+		extra       []float64
+	}{
+		{
+			name:        "Divide two positive integers",
+			errExpected: false,
+			want:        2,
+			n1:          6,
+			n2:          3,
+			extra:       nil,
+		},
+		{
+			name:        "Divide by zero",
+			errExpected: true,
+			want:        0,
+			n1:          6,
+			n2:          0,
+			extra:       nil,
+		},
+		{
+			name:        "Divide two negative integers",
+			errExpected: false,
+			want:        3,
+			n1:          -6,
+			n2:          -2,
+			extra:       nil,
+		},
+		{
+			name:        "Divide five numbers",
+			errExpected: false,
+			want:        1,
+			n1:          60,
+			n2:          2,
+			extra:       []float64{3, 5, 2},
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got, err := calculator.Divide(tC.n1, tC.n2, tC.extra)
+			if err != nil && tC.errExpected == false {
+				t.Fatalf("Cannot divide inputs %.2f %.2f %.2f: %s", tC.n1, tC.n2, tC.extra, err)
```

Comment:
Will the `%.2f` verb work if `tC.extra` is not nil?

Reply:
unless I send not a number... or you meant when it is null?